apps/web/src/components/CrossChainVeCakeModal/hooks/useMultichainVeCakeWellSynced.ts

```diff
@@ -0,0 +1,143 @@
+import { publicClient } from 'utils/wagmi'
+import { Address } from 'viem'
+import { useAccount } from 'wagmi'
+import { CROSS_CHIAN_CONFIG } from '../constants'
+
+export const useMultichainVeCakeWellSynced = (
+  targetChainId: ChainId,
+): {
+  isVeCakeWillSync?: boolean | null
+  isLoading: boolean
+} => {
+  const { address: account } = useAccount()
+  const enabled = Boolean(account)
+  const { data, isLoading } = useQuery({
+    queryKey: [account, 'veCakeSyncData', targetChainId],
+
+    queryFn: () => {
+      if (!account) return undefined
+      return getVCakeAndProxyData(account, targetChainId)
+    },
+
+    enabled,
+    refetchInterval: FAST_INTERVAL,
+    staleTime: FAST_INTERVAL,
+  })
+
+  return { isVeCakeWillSync: data, isLoading }
+}
+
+export const useAllMultichainSyncedCount = (): {
+  totalCount?: number
+  syncedCount?: number
+} => {
+  const { address: account } = useAccount()
+  const enabled = Boolean(account)
+  const { data } = useQuery({
+    queryKey: [account, 'AllMultichainSyncedCount'],
+
+    queryFn: () => {
+      if (!account) return undefined
+      return getAllMultichainSyncedCount(account)
+    },
+    enabled,
+    refetchInterval: FAST_INTERVAL,
+    staleTime: FAST_INTERVAL,
+  })
+
+  return { totalCount: data?.totalCount, syncedCount: data?.syncedCount }
+}
+
+export const getAllMultichainSyncedCount = async (
+  address: Address,
+): Promise<{ totalCount?: number; syncedCount?: number }> => {
+  const queryChainList = Object.keys(CROSS_CHIAN_CONFIG) as unknown as ChainId[]
+  const totalCount = queryChainList.length + 1 // add BSC CHAIN to count
+
+  const multichainIsWellSync = await Promise.all(
+    queryChainList.map((chainId) => getVCakeAndProxyData(address, chainId)),
+  )
+  return {
+    totalCount,
+    syncedCount: (multichainIsWellSync?.filter((isSync) => isSync === true)?.length ?? 0) + 1,
+  }
+}
+
+export const getVCakeAndProxyData = async (address: Address, targetChainId: ChainId): Promise<boolean | null> => {
+  try {
+    const targetClient = publicClient({ chainId: targetChainId })
+    const bscClient = publicClient({ chainId: ChainId.BSC })
+
+    const targetTime = Math.floor(Date.now() / 1000) + 60
+
+    const [{ result: userInfo }] = await bscClient.multicall({
+      contracts: [
+        {
+          address: getVeCakeAddress(ChainId.BSC),
+          abi: veCakeABI,
+          functionName: 'getUserInfo',
+          args: [address],
+        },
+      ],
+    })
+
+    const callsResultBsc = await bscClient.multicall({
+      contracts: [
+        {
+          address: getVeCakeAddress(ChainId.BSC),
+          abi: veCakeABI,
+          functionName: 'balanceOfAtTime',
+          args: [address, BigInt(targetTime)],
+        },
+        {
+          address: getVeCakeAddress(ChainId.BSC),
+          abi: veCakeABI,
+          functionName: 'balanceOfAtTime',
+          args: [userInfo?.[2] ?? '0x0000000000000000000000000000000000000000', BigInt(targetTime)],
```

Comment:
is zero address fallback or should we early return if no user info?

Reply:
It's a fallback if userInfo?[2] have no value it also return 0x0000000000000000000000000000000000000000

Reply:
is it necessary to make this contract call if no user info?

Reply:
Ok, remove the fallback, and early return if there's no userInfo